src/coreclr/vm/syncblk.cpp

```diff
@@ -2200,10 +2208,11 @@ SyncBlock *ObjHeader::GetSyncBlock()
                             }
                             else
                             {
+                                threadId = pThread->GetThreadId();
```

Comment:
In the above block maybe this should also be set to -1 similarly to others, to differentiate between not held and orphaned